enhancements/multi-arch/heterogenous-architecture.md

```diff
@@ -0,0 +1,270 @@
+This becomes even more of the case once we introduce ARM support as a significant number of use cases detail a x86 master node talking to ARM worker nodes. While ACM could be an option in some cases it does not cover the main ones
+we are seeing that this area would need to address.
+
+### API Extensions
+
+Upgrades to a heterogenous payload will require a special field either in the infrastructure or the cluster version resource to indicate that the user wants to upload to a heterogenous image. One of these APIs will be affected to include this.
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogenous clusters which are trying to upgrade
+
+### Risks and Mitigations
+
+Releases and upgrade graphs for all architectures have to move simultaneously and cannot be independent. If there is a bug on one arch, it will cause the edge or the release image to be removed for all arches
+
+## Design Details
+
+### RHCOS Support for ARM Azure
+
+The RHCOS pipeline will have to generate Azure disk images for ARM. However as of today, RHEL does not support hyper-v for ARM and that support is landing in RHEL8.6. Hence 4.11 builds of RHCOS for ARM Azure will be based on RHEL8.6.
+It is TBD whether a separate pipeline with early access RHCOS builds will be needed as RHEL8.6 would not have GA'ed before 4.11.
+
+### OCP Build & Release:
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifest listed image with the component images being manifest listed as well.
+All 4 architectures - x86, ARM, Power and Z will be included in the manifest list. Existing arch specific release payloads continue to be delivered through their individual release controllers.
+A new heterogeneous release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads.
+
+`oc adm release new` would be [enhanced][WRKLDS-370] to allow the creation of manifest list release payloads. The manifest list flow would be triggered whenever the CVO image in an imagestream was a manifest list.
+If the CVO image is a standard manifest, the generated release payload will also be a manifest. If the CVO image is a manifest list, the generated release payload would be a manifest list (containing a manifest
+for each arch possessed by the CVO manifest list). In either case, oc adm release new would permit non-CVO component images to be manifest or manifest lists and pass them through directly to the resultant release manifest(s).
+If a manifest list release payload is generated, each architecture specific release payload manifest will reference the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifest lists, the release controller would use quay.io. There will be a new architecture option added to Cincinnati called "multi"
```

Comment:
doesn't the release controller rely on imagestreams?  any implications there, since we can't have manifestlist-based imagestreams?

Reply:
not sure how the hacked payloads were generated, but i thought from the doc that Justin had on the release options, the plan was to use quay.io which i  have updated here.

Reply:
use of imagestreams is about more than which registry the images are hosted on.

you can have imagestreams that point to images on any registry, and the release controller watches imagestreams to determine when images have new versions (and thus a new payload must be built).  But since imagestreams, when pointed to a manifestlist, do not import all the images from the manifestlist, that has implications for detecting when the images have changed, if the imagestream is the mechanism for identifying which images have changed (or getting the SHAs of the images)

Reply:
ah..i see..ok. @jupierce any idea what the trigger for builds by the heterogeneous release controller will be?

Reply:
since homogeneous payloads will still be manifest(not manifestlist) based, it may be possible to trigger a heterogeneous payload to be built whenever a new payload is produced for any of the included architectures.

that would not require the release controller to watch manifestlist based imagestreams, at least for triggering purposes.  I'm not sure if there are other things it needs to do that are impacted.